### DISCUSSION TITLE
Open access LLMs

### DISCUSSION THREAD

**Discussion by Tenfleques** [lifan0127/ai-research-assistant#8] [Ideas]
In the spirit of research, would it be idea if the assistant can connect to any other LLM even local?

**Comment by lifan0127** [comment]
Behind the scene, this plugin uses LangChain JS to interact with a chat model. So far the only choices in this family are OpenAI and Anthropic, but I imagine more open-access LLMs will become available in the future.

https://js.langchain.com/docs/modules/models/chat/integrations

**Reply by jmgoguikian** [reply]
How did you decide to go with OpenAI fam vs. Anthropic for Zoro RA?

**Reply by lifan0127** [reply]
Hi @jmgoguikian, right now we use the default OpenAI chat model. I personally do not have deep experience with the Anthropic model/api.

**Comment by juniorbonner** [comment]
It would be good to be able to choose to use gpt-4

**Comment by man00ka** [comment]
Local LLMs seem to be possible now, as, for example, ChatOllama is listed by now: https://js.langchain.com/docs/integrations/chat/

Would be really great if this worked in the near future within ARIA : )

**Comment by lifan0127** [comment]
Hi @adkiem, ARIA heavily uses GPT's [function calling feature](https://platform.openai.com/docs/guides/function-calling) to interact with Zotero APIs. I see that open-source LLMs such as Ollama has started to support [similar features](https://python.langchain.com/docs/integrations/chat/ollama_functions) but we will need to measure their performance in comparison to GPT-4. To that end, I am curating a dataset to support such functional testing so we can more easily explore/validate different model configurations.

**Comment by BradKML** [comment]
LiteLLM, OpenRouter, Ollama. The trinity of local use + semi-local services.
